### ISSUE TITLE
Allow GeoNode Styles, Proxy and GetCapa endpoints supporting *any* workspace

### ISSUE THREAD

**Issue by afabiani** [GeoNode/geonode#6454]
This is a preparatory work for a possible real multitenancy implementation.

Currently, by adding to GeoServer (and then to GeoNode) layers belonging to a "workspace" different from the default one, leads to several minor issues around due to the assumption, in most parts of the code, that the workspace is always the default one.

In order to start improving GeoNode and also make it support layers from GeoServer with different workspaces, we should carefully review all the parts of the code where such an assumption has been made and, if possible, remove it.

Long story short, let's remove "single workspace binding" whenever is possible from the codebase, and let's use wildcards instead.

**Comment by gannebamm** [comment]
@afabiani Do you think this is doable for a 3.1 release in terms of development time needed?
